.github/workflows/build_docs.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build Documentation
+
+on:
+  pull_request:
+    paths:
+      - 'docs/**'
+    types:
+      - closed
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Set up Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Build Documentation
+        run: |
+          make docs
+
+      - name: Deploy documentation
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./docs/site
```

Comment:
I suggest publishing it to a dedicated branch: https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-set-another-github-pages-branch-publish_branch

So it does not mess with the git history on the main one. You can change the GitHub's page source on the settings.

Reply:
this action does use a dedicated branch named "gh-pages". It doesn't introduce any build artifacts into the main branch.